src/subpages/list.tsx

```diff
@@ -1,259 +1,133 @@
-          }
 
-          const txNumbers: number[] | null =
-            data && data.blockQuery && data.blockQuery.blocks
-              ? data.blockQuery.blocks.map(block => block!.transactions.length)
-              : null;
-          let totalTxNumber = 0;
-          if (txNumbers) {
-            totalTxNumber = txNumbers.reduce((a, b) => a + b, 0);
-          }
+          const timeTaken: IColumn = {
+            key: 'columnTimeTaken',
+            name: 'Time Taken',
+            minWidth: 50,
+            maxWidth: 200,
+            ...commonProps,
+            isSortedDescending: true,
+            data: 'string',
+            isPadded: true,
+            onRender: (block: Block, index: number) => {
+              if (blocks === null) throw Error('blocks is null');
+              // FIX TODO: beforeBlock index is not available
+              const beforeBlock =
+                blocks[Math.min(index! + 1, block.length - 1)];
+              const beforeTimestamp = Date.parse(beforeBlock.timestamp);
+              const nowTimestamp = Date.parse(block.timestamp);
+              return <>{(nowTimestamp - beforeTimestamp) / 1000}</>;
+            },
+          };
           return (
             <>
-              <div className="cards">
-                <div className="card" key="interval">
-                  <strong>{interval}</strong> sec
-                  <p>Average interval in this page</p>
-                </div>
-                <div className="card" key="difficulty">
-                  <strong>{difficulty.toLocaleString()}</strong>
-                  <p>Average difficulty in this page</p>
-                </div>
-                <div className="card" key="total-tx-number">
-                  <strong>{Number(totalTxNumber).toLocaleString()}</strong>
-                  <p>Total txs in this page</p>
-                </div>
-              </div>
-              <div className="nav">
-                <DefaultButton
-                  onClick={newerHandler}
-                  disabled={loading || offset < 1}
-                  className={css`
-                    margin-right: 5px;
-                  `}>
-                  &larr; Newer
-                </DefaultButton>
-                <DefaultButton disabled={loading} onClick={olderHandler}>
-                  Older &rarr;
-                </DefaultButton>
-              </div>
-              <BlockList
-                blocks={
-                  loading
-                    ? []
-                    : (data!.blockQuery!.blocks as NonNullable<Block[]>)
-                }
-                loading={loading}
+              <SummaryCards blocks={blocks} />
+              <OffsetSwitch
+                olderHandler={olderHandler}
+                newerHandler={newerHandler}
+                disable={{ older: loading || offset < 1, newer: loading }}
               />
+              <BlockList blocks={blocks} loading={loading} columns={columns} />
```

Comment:
So I excluded the `timeTaken` column. instead of below line.

```suggestion
              <BlockList blocks={blocks} loading={loading} columns={[...columns, timeTaken]]} />
```